### PR TITLE
Add Features & Analysis to Secondary Columns

### LINKED ISSUE #5701: Add Features & Analysis to Secondary Column
### Is your feature request related to a problem? Please describe.
Add Features & Analysis to the Secondary Column, following the merge of https://github.com/bbc/simorgh/pull/5417. Features & Analysis will be rendered using a similar pattern to CPS Related Content, where `content` below is Features & Analysis data:

```jsx
<StoryPromoUl>
  {content
    .map(item => (
      <StoryPromoLi key={item.id || item.uri}>
        <StoryPromo item={item} />
      </StoryPromoLi>
    ))}
</StoryPromoUl>;

```

### Describe the solution you'd like
1. Create a new FeaturesAnalysis container. 
2. The Features and Analysis end-point is the front page. The data comes from the CPS group 'Features', which is held on the front page.
E.g. for the Nepali service, https://ares-broker.api.bbci.co.uk/api/index/nepali/front_page
```
content: {
groups: [
type: "feature-main",
title: "Features",
```
This data will need to pass the relevant properties to a `StoryPromo` component as well as to a section label.

3. Use `psammead-story-promo-list` in a similar way shown in the example above to render the dummy fixture data in a secondary column component container.
4. Use `psammead-section-label` to create a "Features & Analysis" section label from the fixture value created in step 2.
5. A11y - This should be wrapped in a section html element with a role region attribute which is labelled via an aria-labelledby attribute which maps to the h2. Below is an example from a front page section

Screen Shot 2020-03-02 at 11 56 12

6. Snapshot tests.

### Describe alternatives you've considered
n/a

### Testing notes
Given I use a screen reader
When I navigate to the features and analysis section on an STY page
Then a region landmark must be announced e.g. Region
And this must be announced with a label 
And the label must be the same as the h2 (e.g. Features and analysis) and be unique to the page
And this should not be inside the main landmark or any other landmark

Given I am using a screenreader
When I navigate to a features and analysis component promo
And there is no content within the promo below the link
Then the promo should not contain heading semantics

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

### Additional context
- https://paper.dropbox.com/doc/Simorgh-WS-STY-onward-journeys--AvIEmS7B0c8VTK8ym0qb6uTrAg-gSSePsYXjWttnHdEIwWYI
 - https://www.bbc.com/thai/thailand-51736465
- The component header is also taken from the front page data. See #5794. The CPS group has a semantic group name with a translation. E.g. for Nepali (https://ares-broker.api.bbci.co.uk/api/index/nepali/front_page) the CPS group for Features and Analysis is 'Features'. This group has a group.strapline.name of "अन‌ि यो पनि", so the heading for the Features and Analysis onward journey on the STY page is "अन‌ि यो पनि".
```
strapline: {
name: "अन‌ि यो पनि"
},
semanticGroupName: "Features"
```

### PR DESCRIPTION
**Pull request by mykeels**
Resolves #5701

**Overall change:** 
_Add a FeaturesAnalysis container, which uses fixtures data to render a Features & Analysis section._

**Code changes:**
- _Added a FeaturesAnalysis container._
- _Added sample fixtures data files (LTR and RTL)._
- _Added stories and unit tests._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by mykeels** [comment]
image

Note the "Features Analysis" section on the right. You will notice the title has a white background. This is because the `SectionLabel` component used, is the same as used in this related content, just before the footer:

image

We'll need a way to pass the right style to it.

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap [hunk 1]
```diff
@@ -0,0 +1,317 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`CpsRelatedContent should render Story Feature components in Live environment 1`] = `
+<DocumentFragment>
+  <section
+    aria-labelledby="related-content-heading"
```
Comment (greenc05) [inline src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: incorrect name for this "related-content-heading", should be "features-analysis-heading"?

Comment (greenc05) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: I see you've update it now.... thanks

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+                dir="ltr"
+              >
+                <div
+                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
+                >
+                  <div
+                    class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
+                  >
+                    <div
+                      class="lazyload-placeholder"
```
Comment (greenc05) [inline src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: should we have an example of an image with alt text??

**Comment by greenc05** [comment]
Will this have a grey background as per your screenshot? https://github.com/bbc/simorgh/pull/5751#issuecomment-593971157

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                  <div
+                    class="ImageOverlayWrapper-sc-1dvfmi3-2 dqBekx"
+                  />
+                </div>
+              </div>
+              <div
+                class="TextGridItem-sc-1lkbq5i-0 jJWtYv"
+                dir="ltr"
+              >
+                <h3
```
Comment (greenc05) [inline src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: Just realised that we shouldn't have headings (h3's) for each of the promos, as there is no related content below them and this fails an accessibility standard. Is it possible to change this to a div/span please?

**Comment by mykeels** [comment]
https://github.com/bbc/simorgh/pull/5751#issuecomment-594024646

@greenc05 I thought they'd all have grey backgrounds. I now see that might be because the sections with grey backgrounds are placeholders. Is that what's happening?

**Comment by greenc05** [comment]
> [#5751 (comment)](https://github.com/bbc/simorgh/pull/5751#issuecomment-594024646)
> 
> @greenc05 I thought they'd all have grey backgrounds. I now see that might be because the sections with grey backgrounds are placeholders. Is that what's happening?

I'm not saying they shouldn't have grey backgrounds, I don't know if they should or not, I was just checking.... could you tell me the hex value of the grey please?

**Comment by mykeels** [comment]
https://github.com/bbc/simorgh/pull/5751#issuecomment-594402076

```
#ECEAE7
```

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: This resolves to something like:
```html
<img 
srcset="https://ichef.bbci.co.uk/news/70/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 70w, https://ichef.bbci.co.uk/news/95/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 95w, https://ichef.bbci.co.uk/news/144/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 144w, https://ichef.bbci.co.uk/news/183/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 183w, https://ichef.bbci.co.uk/news/240/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 240w, https://ichef.bbci.co.uk/news/320/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 320w, https://ichef.bbci.co.uk/news/624/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 624w" 
alt="Some people at the NTA awards" 
src="https://ichef.bbci.co.uk/news/660/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg" 
sizes="(max-width: 1008px) 33vw, 237px" 
width="976" 
class="StyledImg-sc-7vx2mr-0 hIxkbt">
```

**Comment by greenc05** [comment]
Thanks for the colour. FYI there are some colour contrast issues, I've asked UX for a solution

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+const StyledSectionLabel = styled(SectionLabel)`
+  margin-top: 0;
+
+  span {
+    background-color: transparent;
+  }
+`;
+
+const FeaturesAnalysis = ({ content, enableGridWrapper }) => {
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.jsx]: Do we need `enableGridWrapper` for this component? I think it was only needed for `CpsRelatedContent` because it was integrated differently in `MediaAsset` pages compared to `Story` pages.

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    enableGridWrapper ? (
+      <GhostGrid {...a11yAttributes}>
+        <Wrapper>{children}</Wrapper>
+      </GhostGrid>
+    ) : (
+      <Wrapper {...a11yAttributes}>{children}</Wrapper>
+    );
+  FeaturesAnalysisWrapper.propTypes = {
+    children: node.isRequired,
+  };
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.jsx:41]: Do we need propTypes on internal components?

File: src/app/containers/FeaturesAnalysis/index.test.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    expect(document.querySelectorAll(`li[class^='StoryPromoLi']`).length).toBe(
+      features.length,
+    );
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should render Story Feature components in Live environment', () => {
+    const { asFragment } = renderFeaturesAnalysis({
+      bbcOrigin: 'https://www.bbc.co.uk',
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.test.jsx]: How does the change of origin make this and the test above behave differently?

File: src/app/containers/FeaturesAnalysis/index.test.jsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    });
+
+    // x_candy_override should not be used in the live environment
+    expect(document.querySelector(`[href*='x_candy_override']`)).toBeNull();
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should have a "region" role (a11y)', () => {
+    renderFeaturesAnalysis();
+    expect(document.querySelectorAll(`[role='region']`).length).toBe(1);
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.test.jsx]: I think we need an additional test to cover this requirement: 
>This should be wrapped in a section html element with a role region attribute which is labelled via an aria-labelledby attribute which maps to the h2

Specifically the `labelledby` attribute part

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  return (
+    <FeaturesAnalysisWrapper>
+      <Wrapper>
+        <StyledSectionLabel
+          script={script}
+          service={service}
+          dir={dir}
+          labelId="features-analysis-heading"
+        >
+          Featured &amp; Analysis {/** this should be translated */}
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.jsx]:
```suggestion
          Features &amp; Analysis {/** this should be translated */}
```
Also are the translations not part of this PR?

**Comment by greenc05** [comment]
https://github.com/bbc/simorgh/pull/5751#issuecomment-594460752 UX have confirmed that the background colour for the secondary column components is incorrect, this should be lunar (#f2f2f2) and will address the colour contrast issues.

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap (hunk 3 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: Yes, this has been overridden to make it a DIV.

File: src/app/containers/FeaturesAnalysis/index.test.jsx (hunk 6 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.test.jsx]: it doesn't.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 4 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: What does the grid wrapper do?

Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: Seen.

I've taken it out.

**Comment by mykeels** [comment]
### [Preview from March 04, 2020]

After:
- overriding the background color of the H2 title
- changing the H3 links to DIV

---

image

**Comment by mykeels** [comment]
### [Preview from March 05, 2020]

After:
- Changing the Secondary Columns' background color from C_GHOST to C_LUNAR.

image

File: src/app/containers/FeaturesAnalysis/index.test.jsx (hunk 7 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.test.jsx]: Added clearer tests

**Comment by lizcameron** [comment]
Confirmation from UX that there should be no background colour for the Features and Analysis component.

File: src/app/pages/StoryPage/index.jsx [hunk 9]
```diff
@@ -100,7 +101,7 @@ const StoryPage = ({ pageData }) => {
   `;
 
   const ComponentWrapper = styled.div`
-    background: ${C_CHALK};
+    background: ${C_LUNAR};
```
Comment (mykeels) [inline src/app/pages/StoryPage/index.jsx]: Remove this line, since the bg should be transparent.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 8 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: Note: Check with @lizcameron

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap (hunk 3 above)
Comment (jamesdonoh) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: @greenc05 the use of h3 appears to be an existing problem with the Related Content component which the work here is heavily based on (e.g. see https://www.bbc.com/pidgin/tori-51728785).

We've just had a discussion and are going to create a separate issue to look at an accessible solution for both Related Content and Features and Analysis in the round, rather than applying an overly-specific fix here.

Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: See https://github.com/bbc/simorgh/issues/5793

**Comment by mykeels** [comment]
## [Preview from March 06, 2020]

After:

- Reverting Headling to H3
- Removing background from Secondary Columns

Note: the Headline now has a line across it. That's from the `Bar` component in `@bbc/psammead--section-label`. There is reason to believe it has always been there, and the background made it hard to see.

image

**Comment by HarveyPeachey** [comment]
> Note: the Headline now has a line across it. That's from the `Bar` component in `@bbc/psammead--section-label`. There is reason to believe it has always been there, and the background made it hard to see.
> 

The bar is intentional. The section label should have a background colour which is the same colour as the foreground document. You might be overriding it by accident maybe? 

This is what it should look like:
Screenshot 2020-03-06 at 14 03 09

**Comment by mykeels** [comment]
### [Preview from March 09, 2020]

- Make the Features & Analysis Header text have the same bg as the document's body.

image

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 8 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: This is being tracked in https://github.com/bbc/simorgh/issues/5794, which I have assigned to myself, and intend to complete in conjunction with https://github.com/bbc/simorgh/issues/5765

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+const StyledSectionLabel = styled(SectionLabel)`
+  margin-top: 0;
+
+  span {
+    background-color: transparent;
+  }
+
+  span[dir] {
+    background-color: ${C_GHOST};
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.jsx]: Why does the background colour change dependent on the presence of a `dir` attribute?

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 8 above)
Comment (andrewscfc) [reply src/app/containers/FeaturesAnalysis/index.jsx]: Thanks for addressing the translation point, but placeholder text still has a typo we can correct for now 🙂

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+FeaturesAnalysis.propTypes = {
+  // We Reuse the front page story item blocks
+  // Both pages use CPS, so the data schema is the same
+  // This can be found under Frontpage data payloads: content.groups.find(group => group.type === 'feature-main').items
+  content: arrayOf(shape(storyItem)),
+};
+
+FeaturesAnalysis.defaultProps = {
+  content: featuresAnalysis, // @TODO: rm this
```
Comment (andrewscfc) [inline src/app/containers/FeaturesAnalysis/index.jsx:73]: Is todo before this is merged?

**Review by andrewscfc — COMMENTED** [review]
Few more comments

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 11 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx:73]: No, this is to be done in the next PR, that connects the features&analysis data to this component.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 10 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: The DOM structure of the Secondary Column Header is:

```html
<h2>
  <span>
    <span dir="...">...</span>
  </span>
</h2>
```

By default the `h2 span` is a gray color, which we do not need, so we make it transparent.

```
span {
    background-color: transparent;
  }
```

But then, there's a gray bar that runs through the text to the right. We need the actual text's background to be white, so the bar is just visible after it. 

That's why we use `span[dir]` to override the bg color for just the span that contains text.

Comment (andrewscfc) [reply src/app/containers/FeaturesAnalysis/index.jsx]: Oh okay, so its a bit of a work around the Secondary Column Header styling, just not that obvious from looking at it. Not sure what the best way to make this clearer is, kinda feel like there should be a prop on `SectionLabel` to make it's background transparent but not sure that's worth the effort tbh.

**Review by andrewscfc — APPROVED** [review]
Thanks for going through my comments 🙂

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <FeaturesAnalysisWrapper>
+      <Wrapper>
+        <StyledSectionLabel
+          script={script}
+          service={service}
+          dir={dir}
+          labelId="features-analysis-heading"
+        >
+          Featured &amp; Analysis {/** this should be translated */}
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx]: - Can we remove this comment? They can lead to strange behaviour in JSX.
- "Featured" should be corrected to "Features" as per @andrewscfc's review.

File: src/app/containers/FeaturesAnalysis/index.stories.jsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+import { RequestContextProvider } from '#contexts/RequestContext';
+
+const getFeaturesAnalysis = platform => ({ service, dir, data }) => (
+  <div dir={dir}>
+    {/* The above simulates dir being added at the page level */}
+    <ServiceContextProvider service={service}>
+      <RequestContextProvider
+        bbcOrigin="https://www.test.bbc.com"
+        isAmp={platform === 'amp'}
+        pageType="STY" /* Can also be one of other CPS pagetypes */
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.stories.jsx]: I'm going to ignore these comments as they're in a Story. 😄

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 11 above)
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx:73]: What does this comment mean?

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+        </StoryPromoUl>
+      </Wrapper>
+    </FeaturesAnalysisWrapper>
+  );
+};
+
+FeaturesAnalysis.propTypes = {
+  // We Reuse the front page story item blocks
+  // Both pages use CPS, so the data schema is the same
+  // This can be found under Frontpage data payloads: content.groups.find(group => group.type === 'feature-main').items
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx]: Is this comment necessary? It's clear from the code that's written we're reusing `storyItem` which are propTypes imported from `#models/propTypes/storyItem`.

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+const StyledSectionLabel = styled(SectionLabel)`
+  margin-top: 0;
+
+  span {
+    background-color: transparent;
+  }
+
+  span[dir] {
+    background-color: ${C_GHOST};
+  }
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx]: Why does omitting these styles have no effect? Are they unneeded now that the secondary column doesn't have a grey background?

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+import { GridItemConstrainedLarge } from '#lib/styledGrid';
+import StoryPromo from '../StoryPromo';
+
+const Wrapper = styled(GridItemConstrainedLarge)`
+  margin-bottom: ${GEL_SPACING_DBL};
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
+    margin-bottom: ${GEL_SPACING_TRPL};
+  }
+  /* z-index needs explicitly set as the psammead section label component uses negative z-indices */
+  z-index: 0;
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx]: I'm still unsure why a z-index is being set here. Please can you clarify why this is needed?

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+import { storyItem } from '#models/propTypes/storyItem';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { GridItemConstrainedLarge } from '#lib/styledGrid';
+import StoryPromo from '../StoryPromo';
+
+const Wrapper = styled(GridItemConstrainedLarge)`
+  margin-bottom: ${GEL_SPACING_DBL};
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
+    margin-bottom: ${GEL_SPACING_TRPL};
+  }
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx:22]: - Why are these rules necessary? The secondary column component should take care of margins.
- Also please review the content on mobile, it is overly compressed. You may need to reset some margins and padding there.

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 18]
```diff
@@ -0,0 +1,90 @@
+import React, { useContext } from 'react';
+import { arrayOf, shape, node } from 'prop-types';
+import SectionLabel from '@bbc/psammead-section-label';
+import { C_GHOST } from '@bbc/psammead-styles/colours';
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx]: This is an unused import.

File: src/app/containers/FeaturesAnalysis/index.jsx [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+const FeaturesAnalysis = ({ content }) => {
+  const { script, service, dir } = useContext(ServiceContext);
+  const a11yAttributes = {
+    as: 'section',
+    role: 'region',
+    'aria-labelledby': 'features-analysis-heading',
+  };
+  const FeaturesAnalysisWrapper = ({ children }) => (
+    <Wrapper {...a11yAttributes}>{children}</Wrapper>
+  );
```
Comment (simonsinclair) [inline src/app/containers/FeaturesAnalysis/index.jsx:38]: Why does this warrant a new component? Can we set `a11yAttributes` on line 56?

**Review by simonsinclair — COMMENTED** [review]
This is a big piece of work, Michael. 👍 Please see my comments below.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 15 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: They were added after the fact, actually. See my explanation to @andrewscfc https://github.com/bbc/simorgh/pull/5751#discussion_r390141884

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 16 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: This was copied along with the code from `CpsRelatedContent` container. I'm not sure that it's necessary for this component though.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 19 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx:38]: This style was borrowed from `CpsRelatedContent` container.

In that case, the relevant code is:

```jsx
const RelatedContentWrapper = ({ children }) =>
    enableGridWrapper ? (
      <GhostGrid {...a11yAttributes}>
        <Wrapper>{children}</Wrapper>
      </GhostGrid>
    ) : (
      <Wrapper {...a11yAttributes}>{children}</Wrapper>
    );
```

It's clear that it needs to decide between wrapping with `<GhostGrid />` or not, but it's not clear why `<Wrapper />`, which the `RelatedContentWrapper` styles, is still included in its descendants. 

Perhaps, this has some effect on a11y I don't know about?

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 11 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx:73]: It means `remove this`. When we switch from using fixture data to using actual frontpage data, the prop should default to an empty array instead.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 12 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: image 😄

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 15 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: You are right. They seem to be redundant. I've removed them.

File: src/app/containers/FeaturesAnalysis/index.jsx (hunk 16 above)
Comment (mykeels) [reply src/app/containers/FeaturesAnalysis/index.jsx]: I've removed the `z-index`.

**Review by simonsinclair — APPROVED** [review]

**Comment by HarveyPeachey** [comment]
**Dev/A11y Swarm**

-https://bbc.github.io/accessibility-news-and-you/accessibility-news-and-testers
  - [x] Axe
  - [x] Design
  - [x] Colour Contrast
  - [x] Colour blindness
  - [x] Windows HCM
  - [x] FF Colour Preferences
  - [x] Text 200%, Page 200%
  - [x] Icons
  - [ ] ~Heading Order~
  - [x] Check accesibility of browser DOM
  - [x] Keyboard Navigation 

**_Screen Readers (IE11) - Harvey/Simon T:_** 
- [x] ZoomText Magnifier/Reader (Steps)
8. When hovering over the image it doesn't read out anything, need to confirm if this should be the case.
9. Speakit Tool reads out all content, also reads out the image alt text.
10. Similar to the speakit tool where it reads out all content and image alt text.
- [ ] Dragon NaturallySpeaking
- [x] JAWS (Steps)
4. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent.
After navigating past one link JAWS pronounces "BLANK"
Summary is being rendered in HTML but not being read out and is also hidden on the page - should this be the case.
- [x] Read&Write (Steps) 
4. "Read by sentence" setting read out both the link text and the hidden summary.
6. Same behaviour where it reads out the summary.
9. The feature is using a Faux block link so it cannot read out the img alt text.
- [x] NVDA (Firefox) - When clicking before the feature it doesn't go to the desired region. 
4. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation. The first link in the example infinitely loops speech when navigated to.
5. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation. 
6. Same bug where it infinitely loops on the first link.
7.  "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation.


**_Mac, iOS, and Android:_** 
- [x] VoiceOver iOS (iPhone/iPad)
5. Only on first image, it reads out alt text, then in english it says "actions availible very sharp - five faces"
8. "Features & Analysis Region" is read out in Spanish then it reads out "Heading level 2 - Features and Analysis" in Spanish.
- [ ] VoiceOver OS (Mac)
- [x] TalkBack (Chrome on Android)

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap (hunk 3 above)
Comment (greenc05) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: Ok, so apologies if my use of the words 'related content' confused things here. If there is no text content below the promo title, then the title should not be a heading, as headings should have related content below them. So in the case of F&A, each li would not have any content below the 'title'. However in the 'Related content' component we have the time stamp below the promo title/heading so technically this is fine as a heading https://github.com/bbc/bbc-a11y/blob/master/lib/standards/tests/contentMustFollowHeadings.js

Comment (greenc05) [reply src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap]: The only other exception that I'm aware of is the onwards journey component in the middle of the article/story body text, this component cannot have any headings at all due to being in the article/story body and would head up all the article/story content below it, which would not be accurate

File: src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+                class="Headline-sc-1dvfmi3-3 hSJrur"
+              >
+                <a
+                  class="Link-sc-1dvfmi3-5 yQgSu"
+                  href="/igbo/afirika-23252735"
+                >
+                  STY - Ọrụ bekee na ịrụrụ onwe gị ọrụ, kedụ nk?
+                </a>
+              </h3>
+              <p
```
Comment (greenc05) [inline src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap:83]: Lets remove this p tag and it's content please, this would not be there.

**Comment by greenc05** [comment]
> **Dev/A11y Swarm**
> 
> -https://bbc.github.io/accessibility-news-and-you/accessibility-news-and-testers
> 
> * [x]  Axe
> * [x]  Design
> * [x]  Colour Contrast
> * [x]  Colour blindness
> * [x]  Windows HCM
> * [x]  FF Colour Preferences
> * [x]  Text 200%, Page 200%
> * [x]  Icons
> * [ ]  ~Heading Order~
> * [x]  Check accesibility of browser DOM
> * [x]  Keyboard Navigation
> 
> **_Screen Readers (IE11) - Harvey/Simon T:_**
> 
> * [x]  ZoomText Magnifier/Reader (Steps)
> 
> 1. When hovering over the image it doesn't read out anything, need to confirm if this should be the case.
> 2. Speakit Tool reads out all content, also reads out the image alt text.
> 3. Similar to the speakit tool where it reads out all content and image alt text.
> 
> * [ ]  Dragon NaturallySpeaking
> * [x]  JAWS (Steps)
> 
> 1. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent.
>    After navigating past one link JAWS pronounces "BLANK"
>    Summary is being rendered in HTML but not being read out and is also hidden on the page - should this be the case.
> 
> * [x]  Read&Write (Steps)
> 
> 1. "Read by sentence" setting read out both the link text and the hidden summary.
> 2. Same behaviour where it reads out the summary.
> 3. The feature is using a Faux block link so it cannot read out the img alt text.
> 
> * [x]  NVDA (Firefox) - When clicking before the feature it doesn't go to the desired region.
> 
> 1. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation. The first link in the example infinitely loops speech when navigated to.
> 2. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation.
> 3. Same bug where it infinitely loops on the first link.
> 4. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation.
> 
> **_Mac, iOS, and Android:_**
> 
> * [ ]  VoiceOver iOS (iPhone/iPad)
> * [ ]  VoiceOver OS (Mac)
> * [ ]  TalkBack (Chrome on Android)

**ZoomText Magnifier/Reader**
"8. When hovering over the image it doesn't read out anything, need to confirm if this should be the case." @HarveyPeachey This is due to faux block link CSS and expected

**JAWS**
"Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent." @HarveyPeachey were you testing this on Mundo?

"After navigating past one link JAWS pronounces "BLANK" - Fine

"Summary is being rendered in HTML but not being read out and is also hidden on the page - should this be the case." Why are we rendering the summary @mykeels? Is it possible not to please? This is also why it's reading out with Read&Write @HarveyPeachey 

**NVDA**
Sounds a bit buggy, might be advisable to re-test NVDA when we have the translations @HarveyPeachey ?

**Comment by HarveyPeachey** [comment]
> 
> **JAWS**
> "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent." @HarveyPeachey were you testing this on Mundo?
> 
> **NVDA**
> Sounds a bit buggy, might be advisable to re-test NVDA when we have the translations @HarveyPeachey ?

Yeah, I was testing on Mundo as that's where our big Test asset lives. The main point of that statement was that it read out "Features & Analysis Region" and then it also readout "Heading level 2 - Features and Analysis", which was a bit off as it's repeating the component name.

With the NVDA stuff, I agree this should be tested when translations are added, I'll make this into a seperate issue :).

**Comment by mykeels** [comment]
> "Why are we rendering the summary @mykeels? Is it possible not to please? This is also why it's reading out with Read&Write @HarveyPeachey"

On a medium-sized screen, the summary is displayed. 
image
